pandas/core/generic.py

```diff
@@ -454,22 +454,29 @@ def _get_axis_resolvers(self, axis):
         d[axis] = dindex
         return d
 
-    def _get_index_resolvers(self):
-        d = {}
+    def _get_index_resolvers(self) -> Dict[str, ABCSeries]:
+        from pandas.core.computation.parsing import clean_column_name
+
+        d: Dict[str, ABCSeries] = {}
         for axis_name in self._AXIS_ORDERS:
             d.update(self._get_axis_resolvers(axis_name))
-        return d
 
-    def _get_space_character_free_column_resolvers(self):
-        """Return the space character free column resolvers of a dataframe.
+        return {clean_column_name(k): v for k, v in d.items() if k is not int}
```

Comment:
should this be `if not isinstance(k, int)`?  this is causing mypy failures in #30694.

Reply:
yes that looks like it needs changing

Reply:
Oeh, that's really embarrassing. Yes, it should be `if not isinstance(k, int)`.

What should be best to fix this. Should I make a PR for this?

Reply:
#30698